src/book/04-guides/01-nubots/02-camera-calibration.mdx

```diff
@@ -0,0 +1,74 @@
+title: Camera Calibration
+description: Learn how to calibrate cameras.
+slug: /guides/nubots/camera-calibration
+---
+
+Camera calibration is done to determine optimal parameters for the cameras, such as focal length and the transform from one camera to the other. More on camera calibration can be found on [the vision page](/system/nubots/vision).
+
+The calibration is done in two steps. The first is creating the nbs recording and the second is using that nbs recording to find the camera parameters.
+
+1. Change `cameratest.role` to include `support::logging::DataLogging`
+
+   ```bash
+   nuclear_role(
+     # FileWatcher, ConsoleLogHandler and Signal Catcher Must Go First. KinematicsConfiguration usually goes after these
+     # and without it many roles do not run
+     extension::FileWatcher
+     support::SignalCatcher
+     support::logging::ConsoleLogHandler
+     support::logging::DataLogger
+     # This must come first as it emits config which many roles depend on (e.g. SensorFilter, WalkEngine)
+     motion::KinematicsConfiguration
+     support::configuration::NetworkConfiguration
+     support::configuration::GlobalConfig
+     platform::darwin::HardwareIO
+     platform::darwin::SensorFilter
+     input::Camera
+     support::NUsight
+     output::ImageCompressor
+     support::configuration::SoccerConfig
+   )
+   ```
+
+2. Build using `./b build` and then install onto a robot using `./b install n<number>`. More on building can be found on the [Getting Started page](/guides/nubots/getting-started) or the [Build System page](/system/nubots/build-system).
+
+3. On the robot, enable compressed image recording in `/config/DataLogging.yaml`. Do this by setting `message.output.CompressedImage` to `true`.
+
+4. Ensure the Left and Right camera serial numbers are correct in their respective configuration files (`Left.yaml` and `Right.yaml`).
+
+   The serial numbers can be retrieved by running the `sudo lsusb -s <bus number>:<device number> -v` command on the robot.
+
+   Next you will need to figure out which is left and which is right. Change the serial number of one camera (this will ensure it is not found when running cameratest). Enable compressed images in `NUsight.yaml`. Run `./cameratest` and test which camera is working, simply by covering one camera and seeing the resulting image in the NUsight vision tab. Alternatively you could unplug one camera rather than change the serial number. If the serial number is on the wrong camera configuration file, switch them.
+
+5. Check the focus of the lenses using the NUsight vision tab. If it is not sharp, focus the camera.
+
+6. Check the transform matrix from the left camera to the pitch motor (Hpc) is correct in the `Left.yaml` file.
+
+7. Run `./cameratest` on the robot and hold up a checkerboard, moving it around the area to cover most of the view over time. Make sure you cover all of the available image, and make sure that you tilt the board at different angles from the lens.
```

Comment:
You should also mention that they should do different distances and since the lenses are 180º fov, you should move the board quite close to the lenses to cover a larger amount of the view